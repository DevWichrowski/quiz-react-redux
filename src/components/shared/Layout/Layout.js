import React from 'react';
import "./Layout.scss"
import Header from "../Header/Header";
import Footer from "../Footer/Footer";

const Layout = ({children}) => {
    return (
        <div className="layout">
            <Header className="layout__main-header" navigation={"/"}/>>
            <div className="layout__content">
                {children}
            </div>
            <Footer/>
        </div>
    );
};

export default Layout;