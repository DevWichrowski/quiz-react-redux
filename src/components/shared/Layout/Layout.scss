@import "./src/styles/colors";
@import "./src/styles/mixins";

.layout {

  &__content {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    border-radius: 10px;
    @include boxShadow();
    width: 60%;
    background-color: white;
    height: 60vh;
  }

  @media screen and (max-width: 1000px) {
    &__content {
      width: 80%;
    }
  }

  @media screen and (max-width: 600px) {
    &__content {
      width: 90%;
    }
  }

  &__main-header {
    width: 100%;
    height: 80px;
    background-color: $gray;
    font-size: 30px;
    display: flex;
    justify-content: center;
    font-weight: bolder;
    align-items: center;
    @include boxShadow();
    color: $blue;
  }
}


