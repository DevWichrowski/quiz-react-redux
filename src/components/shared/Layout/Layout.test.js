import {testComponentRender} from "../../../core/helpers/tests/test-component-render";
import React from "react";
import Layout from "./Layout";


describe('Layout', () => {
    testComponentRender(<Layout>Test</Layout>)
});
