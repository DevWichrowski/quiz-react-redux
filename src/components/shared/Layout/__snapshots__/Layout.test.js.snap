// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render 1`] = `
<div
  className="layout"
>
  <Header
    className="layout__main-header"
    homeNavigation={true}
  />
  <div
    className="layout__content"
  >
    Test
  </div>
  <Footer />
</div>
`;
